yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+      }
+      enum "longer-prefix" {
+	value 5;
+	description
+	  "Lookup both unicast and mcast, use entry with longer prefix.";
+      }
+    }
+    description
+      "Multicast RPF lookup behavior";
+  }
+
+  typedef nexthop-type {
+    type enumeration {
+      enum "ifindex" {
+	value 1;
+	description
+	  "Specific interface.";
+      }
+      enum "ip4" {
+	value 2;
+	description
+	  "IPv4 address.";
+      }
+      enum "ip4-ifindex" {
+	value 3;
+	description
+	  "IPv4 address and interface.";
+      }
+      enum "ip6" {
+	value 4;
+	description
+	  "IPv6 address.";
+      }
+      enum "ip6-ifindex" {
+	value 5;
+	description
+	  "IPv6 address and interface.";
+      }
+      enum "blackhole" {
+	value 6;
+	description
+	  "Unreachable or prohibited.";
+      }
+    }
+    description
+      "Nexthop types.";
+  }
+
+  typedef blackhole-type {
+    type enumeration {
+      enum "unspec" {
+	value 0;
+	description
+	  "Generic unreachable.";
+      }
+      enum "null" {
+	value 1;
+	description
+	  "Null type.";
+      }
+      enum "reject" {
+	value 2;
+	description
+	  "ICMP unreachable.";
+      }
+      enum "prohibited" {
+	value 3;
+	description
+	  "ICMP admin-prohibited.";
+      }
+    }
+    default "null";
+    description
+      "Nexthop blackhole types.";
+  }
+
+  typedef gateway-address {
+    type union {
+      type inet:ipv4-address;
+      type inet:ipv6-address;
+    }
+  }
+
+  /*
+   * Nexthop object
+   */
+  grouping frr-nexthop {
+    leaf nh-type {
+      type frr-zebra:nexthop-type;
+      mandatory true;
+      description
+	"The nexthop type.";
+    }
+
+    leaf gateway {
+      type frr-zebra:gateway-address;
+      description
+	"The nexthop gateway address.";
+    }
+
+    leaf vrf {
+      type string;
+      description
+	"The nexthop vrf name, if different from the route.";
+    }
+
+    leaf interface {
+      type string;
+      description
+	"The nexthop egress interface.";
+    }
+
+    leaf bh-type {
+      type frr-zebra:blackhole-type;
+      description
+	"A blackhole sub-type, if the nexthop is a blackhole type.";
+    }
+
+    leaf flags {
+      type uint32;
+      description
+	"The nexthop's raw flags value.";
+    }
+
+    leaf is-duplicate {
+      type empty;
+      description
+	"Duplicate nexthop; will be ignored.";
+    }
+    leaf is-recursive {
+      type empty;
+      description
+	"Nexthop must be resolved through another gateway.";
+    }
+    leaf is-onlink {
+      type empty;
+      description
+	"Nexthop is directly connected.";
+    }
+    leaf is-active {
+      type empty;
+      description
+	"Nexthop is active.";
+    }
+
+    uses rt-types:mpls-label-stack {
+      description
+	"Nexthop's MPLS label stack.";
+    }
+
+    leaf mtu {
+      type uint32;
+      description
+	"The nexthop's specific MTU.";
+    }
+
+  } // End of nexthop
+
+  /*
+   * Nexthop-group container
+   */
+  grouping frr-nexthop-group {
```

Comment:
this should be a common bit as that it is not unique to zebra?

Reply:
Yes, agreed - I'll move the nexthop info to a more shareable module.